### PR TITLE
Revise space_to_batch

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - add backend tests
 - add SLT tests
 - add Serialization SLT tests
 - add visitors test
 - add type_prop tests
 - refactor ngraph operator
### Tickets:
 - 44627

### CODE REVIEW

**Comment by bszmelcz** [comment]
@jdanieck  for some configs of the space_to_batch_5x5 test from backend/space_to_batch.in.cpp there were segmentation faults. Because of the lack of time I did not find out the reason for that behaviour, but I swapped inputs in failing test to 4x4 and it seems like seg faults are gone. At the moment I see that CI is still failing, but I do not know if the problem is caused by the code or by CI itself.

File: ngraph/test/type_prop/space_to_batch.cpp [hunk 1]
```diff
@@ -110,3 +110,104 @@ TEST(type_prop, space_to_batch_dynamic_shape_dynamic_rank)
...
+    try
+    {
+        auto space_to_batch =
+            make_shared<op::v1::SpaceToBatch>(data, block_shape, pads_begin, pads_end);
+        // Input element type is float32
+        FAIL() << "Invalid f32 element type for pads_end not detected";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), "crops_end must be an integral number but got");
```
Comment (jdanieck) [inline ngraph/test/type_prop/space_to_batch.cpp:180]: Copy-paste error `crops_end` -> `pads_end`

File: ngraph/test/type_prop/space_to_batch.cpp [hunk 2]
```diff
@@ -110,3 +110,104 @@ TEST(type_prop, space_to_batch_dynamic_shape_dynamic_rank)
...
+    try
+    {
+        auto space_to_batch =
+            make_shared<op::v1::SpaceToBatch>(data, block_shape, pads_begin, pads_end);
+        // Input element type is float32
+        FAIL() << "Invalid f32 element type for pads_begin not detected";
+    }
+    catch (const NodeValidationFailure& error)
+    {
+        EXPECT_HAS_SUBSTRING(error.what(), "crops_begin must be an integral number but got");
```
Comment (jdanieck) [inline ngraph/test/type_prop/space_to_batch.cpp:155]: Copy-paste error `crops_begin` -> `pads_begin`

File: ngraph/test/type_prop/space_to_batch.cpp (hunk 1 above)
Comment (jdanieck) [reply ngraph/test/type_prop/space_to_batch.cpp]: Fixed in this [commit](https://github.com/openvinotoolkit/openvino/pull/6034/commits/367dd5c19b1b7bc4fe7482a2d083276483a1520e).

File: ngraph/test/type_prop/space_to_batch.cpp (hunk 2 above)
Comment (jdanieck) [reply ngraph/test/type_prop/space_to_batch.cpp]: Fixed in this [commit](https://github.com/openvinotoolkit/openvino/pull/6034/commits/367dd5c19b1b7bc4fe7482a2d083276483a1520e).

**Review by jdanieck — APPROVED** [review]

**Comment by jdanieck** [comment]
@tomdol maybe you'll have time to review & merge to offload @ilyachur ? Unreviewed PR's started to pile up because of the release.

**Review by tomdol — APPROVED** [review]

**Pull request merged**